### PR TITLE
Course Material Blocks

### LINKED ISSUE #4: Implement Common Blocks
https://wordpress.org/support/article/blocks/

- Paragraph
    - [x] Rendering
    - [x] Attributes
- Image
    - [x] Rendering
    - [x] Attributes
- Heading
    - [x] Rendering
    - [x] Attributes
- Gallery
    - [x] Rendering
    - [x] Attributes
- List
    - [x] Rendering
    - [x] Attributes
- Quote
    - [x] Rendering
    - [x] Attributes
- Audio
    - [x] Rendering
    - [x] Attributes
- Cover
    - [x] Rendering
    - [x] Attributes
- File
    - [x] Rendering
    - [x] Attributes
- Video
    - [x] Rendering
    - [x] Attributes

### LINKED ISSUE #5: Implement Formatting Blocks
https://wordpress.org/support/article/blocks/

- Code
    - [x] Rendering
    - [x] Attributes
- Classic
    - [x] Rendering
    - [x] Attributes
- Custom HTML
    - [x] Rendering
    - [x] Attributes
- Preformatted
    - [x] Rendering
    - [x] Attributes
- Pullquote
    - [x] Rendering
    - [x] Attributes
- Table
    - [x] Rendering
    - [x] Attributes
- Verse
    - [x] Rendering
    - [x] Attributes

### LINKED ISSUE #6: Implement Layout Element Blocks
https://wordpress.org/support/article/blocks/

- Page Break
    * [x] Disable
- Spacer
    * [x] Rendering
    * [x] Attributes
- Buttons (new 5.4)
    * [x] Rendering
    * [x] Attributes
- Button (deprecated)
    * [x] Disable
- Columns
    * [x] Rendering
    * [x] Attributes
- Group
    * [x] Rendering
    * [x] Attributes
- Media & Text
    * [x] Rendering
    * [x] Attributes
- More
    * [x] Rendering
    * [x] Attributes
- Separator
    * [x] Rendering
    * [x] Attributes

### PR DESCRIPTION
**Pull request by sebazai**
Resolves #4 
Resolves #5 
Resolves #6

### CODE REVIEW

File: services/course-material/src/styles/colorMapper.tsx [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-  "pale-pink": "#FFC0CB",
-  "luminous-vivid-orange": "#FF7F50",
-  "luminous-vivid-amber": "#FFBF00",
-  "light-green-cyan": "#00FA9A",
-  "vivid-green-cyan": "#7FFF00",
-  "pale-cyan-blue": "#66CDAA",
-  "vivid-cyan-blue": "#00FFFF",
-  "vivid-purple": "#800080",
+export const colorMap: { [colorName: string]: string | undefined } = {
+  // eslint-disable-next-line prettier/prettier
```
Comment (nygrenh) [inline services/course-material/src/styles/colorMapper.tsx:5]: Should not be a problem, let's go with prettier's preference.

**Review by nygrenh — CHANGES_REQUESTED** [review]

File: services/course-material/src/components/ContentRenderer/core/common/Image/ImageBlock.tsx [hunk 2]
```diff
@@ -72,6 +74,9 @@ const ImageBlock: React.FC<BlockRendererProps<ImageAttributes>> = ({ data }) =>
               src={url}
               alt={alt}
             />
+            {linkTarget === "_blank" && (
```
Comment (nygrenh) [inline services/course-material/src/components/ContentRenderer/core/common/Image/ImageBlock.tsx:77]: contains

File: system-tests/src/tests/course-material/content-components-render.spec.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  await page.waitForSelector("text=100px wide")
+
+  await expectScreenshotsToMatchSnapshots({
+    page,
+    headless,
+    snapshotName: "content-components-renderer-view",
+    waitForThisToBeVisibleAndStable: null,
+    toMatchSnapshotOptions: { threshold: 0.4 },
+    pageScreenshotOptions: { fullPage: true },
+    axeSkip: true,
```
Comment (nygrenh) [inline system-tests/src/tests/course-material/content-components-render.spec.ts:44]: Resolve

**Review by nygrenh — CHANGES_REQUESTED** [review]

File: system-tests/src/tests/course-material/content-components-render.spec.ts (hunk 3 above)
Comment (sebazai) [reply system-tests/src/tests/course-material/content-components-render.spec.ts:44]: Wil lresolve in another task as Background colors for headings color palette is disabled for some reason on the latest version.

**Review by jolampi — APPROVED** [review]

**Pull request merged**